splink/comparison_level_library.py

```diff
@@ -88,6 +106,71 @@ def create_label_for_charts(self) -> str:
         return f"Levenshtein distance of {self.col_name} <= {self.distance_threshold}"
 
 
+class DatediffLevel(ComparisonLevelCreator):
+    def __init__(
+        self,
+        col_name: str,
+        date_threshold: int,
+        date_metric: str = "day",  ##TODO: Lock down to sqlglot supported values
+        cast_strings_to_date: bool = False,
+        date_format: str = "%Y-%m-%d",
+    ):
+        """A comparison level using a date difference function
+
+        e.g. abs(date_diff('day', "mydate_l", "mydate_r")) <= 2  (duckdb dialect)
+
+        Args:
+            col_name (str): Input column name
+            date_threshold (int): The threshold for the date difference
+            date_metric (str): The unit of time ('day', 'month', 'year')
+                for the threshold
+            cast_strings_to_date (bool): Whether to cast string columns to date format
+            date_format (str): The format of the date string
+        """
+        super().__init__(col_name)
+        self.date_threshold = date_threshold
+        self.date_metric = date_metric
+        self.cast_strings_to_date = cast_strings_to_date
+        self.date_format = date_format
+
+    @unsupported_splink_dialects(["sqlite"])
```

Comment:
Do we want this to decorate the class? I think it would make it more explicit that sqlite isn't supported.

Reply:
interesting.  maybe put a question on splink dev to see what others thing.  i think i'm happy either way but possibly have slight preference for decorating the specific method that's unsupported